### ISSUE TITLE
Runner POD restart causes error

### ISSUE THREAD

**Issue by dennybaa** [evryfs/github-actions-runner-operator#186]
Howdy @davidkarlsen , it's not operator relevant I believe. Though pods might end up in crashloop, for easy reproduction I just kill Listner process and the error pops up:

`Cannot configure the runner because it is already configured. To reconfigure the runner, run 'config.cmd remove' or './config.sh remove' first.`

**Comment by davidkarlsen** [comment]
Are your volumes setup as in the example? I believe they will then have same lifecycle as pod and the "already configured" problem should disappear.

**Comment by dennybaa** [comment]
Hi there @davidkarlsen , want to share some updates to you since I use slightly modified containers as well as different volume  setup from yours. So a small intro, github actions runner restarts after update, thus I use emptydir volume for home, in the init container I copy over runner home directory into this...

Actually this error won't happen upon restart if to check for `.credentials` and omit "reconfiguration" (`./config`) everything's become perfect. So my issue is resolved.

Though it would be very interesting to hear your ideas on why you leave the "runner autoupdate" issue beyond the scope in your containers and samples?

**Comment by davidkarlsen** [comment]
I always favour to build immutable containers so they don't rely on stuff in init, especially high-traffic ones like these where there is a lot of spinning up and down, so I prebuild the image with the latest runner in it and simply bump the container if a new version is available. The image will be cached at the hosts so that if a new pod is needed it's already in place.
However the trick to check for .credentials can probably be put in w/o any negative effects. Do you want to do a PR against https://github.com/evryfs/github-actions-runner ?

**Comment by dennybaa** [comment]
Yup will contribute definitely soon. We all favor immutable containers however the reality with the runner leaves no choice... @davidkarlsen Btw what is the delay between the new version appears and the new container publishing takes place, do you perform this automatically? If this is the matter of several minutes say 10 minutes, this is suitable for production but if more then i'm afraid persisting runner updates is a must.

**Comment by davidkarlsen** [comment]
I usually catch them at prerelease as I have a feed on the release-rss — just see history of the repo. We rely on this ourselves. Not automated — but fast.
